server/model/monitor.js

```diff
@@ -488,6 +490,20 @@ class Monitor extends BeanModel {
                     } else {
                         throw new Error("Server not found on Steam");
                     }
+                } else if (this.type === "gamedig") {
+                    try {
+                        const state = await Gamedig.query({
+                            type: this.game,
+                            host: this.hostname,
+                            port: this.port
+                        });
```

Comment:
If you add `givenPortOnly: true` here, you can avoid the multiple query issue I described.